packages/vuetify/src/mixins/overlayable/index.ts

```diff
@@ -146,19 +136,29 @@ export default Vue.extend<Vue & Toggleable & Stackable & options>().extend({
         }
       }
 
-      if (e.target === this.overlay ||
-        (e.type !== 'keydown' && e.target === document.body) ||
-        this.checkPath(e)) e.preventDefault()
+      if (e.target === this.overlay || (e.type !== 'keydown' && e.target === document.body) || this.checkPath(e)) e.preventDefault()
     },
     hasScrollbar (el?: Element) {
       if (!el || el.nodeType !== Node.ELEMENT_NODE) return false
 
       const style = window.getComputedStyle(el)
       return ['auto', 'scroll'].includes(style.overflowY!) && el.scrollHeight > el.clientHeight
     },
-    shouldScroll (el: Element, delta: number) {
-      if (el.scrollTop === 0 && delta < 0) return true
-      return el.scrollTop + el.clientHeight === el.scrollHeight && delta > 0
+    shouldScroll (this: any, el: Element, delta: number) {
```

Comment:
> `this: any`

If you were getting type errors from the recursion you may need to just specify a return type instead